File: tests/human/x/pl/update_stmt.pl
:- initialization(main).
people([
  person('Alice',17,'minor'),
  person('Bob',25,'unknown'),
  person('Charlie',18,'unknown'),
  person('Diana',16,'minor')
]).
update([],[]).
update([person(N,A,S)|T],[person(N,A2,S2)|T2]) :-
    (A >= 18 -> S2='adult', A2 is A+1 ; S2=S, A2=A),
    update(T,T2).
main :-
    people(P), update(P,_Updated),
    writeln('ok'), halt.

File: tests/human/x/ex/update_stmt.ex
defmodule Person do
  defstruct [:name, :age, :status]
end

people = [
  %Person{name: "Alice", age: 17, status: "minor"},
  %Person{name: "Bob", age: 25, status: "unknown"},
  %Person{name: "Charlie", age: 18, status: "unknown"},
  %Person{name: "Diana", age: 16, status: "minor"}
]

people = Enum.map(people, fn p ->
  if p.age >= 18 do
    %{p | status: "adult", age: p.age + 1}
  else
    p
  end
end)

ExUnit.start()

defmodule UpdateTest do
  use ExUnit.Case, async: true
  test "update adult status" do
    assert Enum.at(people, 1) == %Person{name: "Bob", age: 26, status: "adult"}
    assert Enum.at(people, 2) == %Person{name: "Charlie", age: 19, status: "adult"}
  end
end

IO.puts("ok")
